osdk/tools/Dockerfile.tdx

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: MPL-2.0
```

Comment:
I checked some established projects like Spark and PyTorch to see how they have named `Dockefile`. They do not take the naming convention of `Dockerfile.<tag_name>`. Instead, they give the `Dockerfile` for each tag a dedicated directory.

So I suggest re-organizing the `tools` directory as follows.

```
tools/
    docker/
        ubuntu/
            Dockerfile
        ubuntu-tdx/
            Dockerfile
        build_image.sh
        run_container.sh
```